polishNotationCalc.cpp

```diff
@@ -0,0 +1,92 @@
+#include "polishNotationCalc.hpp"
+namespace dev
+{
+    PolishCalc::PolishCalc()
+    {
+        insertFunctionInsideMap();
+    }
+
+    bool PolishCalc::isNumeric(const char num)
+    {
+        double temp = num - '0';
+        if (temp >= 0 && temp <= 9)
+            return true;
+
+        return false;
+    }
+    bool PolishCalc::isOperator(const char num)
+    {
+        if (num == '+' || num == '-' || num == '*' || num == '/')
+            return true;
+        return false;
+    }
+   
+    double PolishCalc::calc( const std::string &str)
+    {
+        if (str.size() == 0)
+            throw Exception{"Nothing to work on as Input is empty"};
+        for (int i = 0; i < str.size(); i++)
+        {
+            if (str[i] == ' ')
+                continue;
+            if (isNumeric(str[i]))
+            {
+                pushOperandInStack(str, i);
+                
+            }
+            if (isOperator(str[i]))
+            {
+                auto operandVal = getOperandFromStack();
+                double res = 0;
+                res = mOperatorMap[str[i]](operandVal.first, operandVal.second);
+                mCalculatorStack.push(res);
+            }
+        }
+
+        if (mCalculatorStack.size() > 1)
+            throw Exception{"Not a valid string"};
+
+        return mCalculatorStack.top();
+    }
+
+    void PolishCalc::pushOperandInStack(const std::string &str, int& curr_pos)
+    {
+        double operand = 0;
+        std::string temp = "";
+        while (curr_pos < str.size() && !(str[curr_pos] == ' '))
```

Comment:
In case of input like `2 4+` , solution behaviour is undefined. 
Try to handle this error. 
Best option is to create prevalidation function and contract for pushOperandInStack.